tests/api/crud/test_alerts.py

```diff
@@ -104,33 +94,198 @@ async def test_validate_alert_name(
         project=project,
         ids=[123],
     )
-    event_kind = mlrun.common.schemas.alert.EventKind.FAILED
-    notification = mlrun.common.schemas.Notification(
-        kind="slack",
-        name="slack_notification",
-        secret_params={
-            "webhook": "https://hooks.slack.com/services/",
-        },
-        condition="oops",
-    )
+    event_kind = alert_objects.EventKind.FAILED
+    alert_trigger = alert_objects.AlertTrigger(events=[event_kind])
 
-    alert_data = mlrun.common.schemas.alert.AlertConfig(
+    alert_data = _generate_alert_data(
         project=project,
         name=alert_name,
-        summary="The job has failed",
-        severity=mlrun.common.schemas.alert.AlertSeverity.MEDIUM,
-        entities=entity,
-        trigger=mlrun.common.schemas.alert.AlertTrigger(events=[event_kind]),
-        notifications=[
-            mlrun.common.schemas.AlertNotification(notification=notification)
-        ],
+        entity=alert_entity,
+        summary=alert_summary,
+        trigger=alert_trigger,
     )
     with expectation:
         server.api.crud.Alerts().store_alert(
-            db, project=project, name=alert_name, alert_data=alert_data
+            session=db,
+            project=project,
+            name=alert_name,
+            alert_data=alert_data,
+        )
+
+
+@pytest.mark.parametrize(
+    "modify_field, modified_value, should_reset",
+    [
+        # Non-functional fields:
+        ("summary", "The job has failed again", False),
+        ("description", "Job failure detected", False),
+        ("severity", alert_objects.AlertSeverity.HIGH, False),
+        (
+            "notifications",
+            [
+                alert_objects.AlertNotification(
+                    notification=mlrun.common.schemas.Notification(
+                        kind="webhook",
+                        name="webhook_notification",
+                        params={
+                            "url": "some-webhook-url",
+                        },
+                    )
+                )
+            ],
+            False,
+        ),
+        ("reset_policy", alert_objects.ResetPolicy.AUTO, False),
+        # Functional fields:
+        (
+            "entities",
+            alert_objects.EventEntities(
+                kind=alert_objects.EventEntityKind.JOB,
+                project="project-name",
+                ids=[456],
+            ),
+            True,
+        ),
+        (
+            "trigger",
+            alert_objects.AlertTrigger(
+                events=[alert_objects.EventKind.DATA_DRIFT_DETECTED]
+            ),
+            True,
+        ),
+        (
+            "criteria",
+            alert_objects.AlertCriteria(
+                count=5,
+                period="10m",
+            ),
+            True,
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "force_reset",
+    [False, True],
+)
+async def test_alert_reset_with_fields_updates(
```

Comment:
I would also test that changing multiple fields simultaneously works as expected